f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py

```diff
@@ -1381,19 +1399,25 @@ def _update_listener_status(self, service):
                         lb_const.OFFLINE)
 
     @log_helpers.log_method_call
-    def _update_loadbalancer_status(self, service):
+    def _update_loadbalancer_status(self, service, timed_out=False):
         """Update loadbalancer status in OpenStack """
-        loadbalancer = service['loadbalancer']
-        provisioning_status = loadbalancer['provisioning_status']
+        loadbalancer = service.get('loadbalancer', {})
+        provisioning_status = loadbalancer.get('provisioning_status',
+                                               plugin_const.ERROR)
 
         if (provisioning_status == plugin_const.PENDING_CREATE or
                 provisioning_status == plugin_const.PENDING_UPDATE):
-            operating_status = (lb_const.ONLINE)
+            if timed_out:
+                operating_status = (lb_const.OFFLINE)
```

Comment:
So, if we fail to provision a member, we know that it is in error.  For a loadbalancer, we need to conditionally set it to an error state if it is PENDING_CREATE.  Otherwise, we let neutron set the status to DEGRADED, i need to add a check here.  Good catch.